packages/wrangler/src/init.ts

```diff
@@ -1167,16 +1168,92 @@ export async function mapBindings(
 							};
 						}
 						break;
-					default: {
-						// If we don't know what the type is, its an unsafe binding
-						// eslint-disable-next-line @typescript-eslint/no-explicit-any
-						if (!(binding as any)?.type) {
-							break;
+					case "secret_text":
+						// Ignore secrets
+						break;
+					case "version_metadata": {
+						{
+							configObj.version_metadata = {
+								binding: binding.name,
+							};
 						}
+						break;
+					}
```

Comment:
Nit: maybe you want this?
```suggestion
					case "version_metadata": {
						configObj.version_metadata = {
							binding: binding.name,
						};
					  break;
					}
```